crates/matrix-sdk/src/oidc/cross_process.rs

```diff
@@ -19,9 +19,28 @@ use super::OidcSessionTokens;
 const OIDC_SESSION_HASH_KEY: &str = "oidc_session_hash";
 
 /// Newtype to identify that a value is a session tokens' hash.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, PartialEq, Eq)]
 struct SessionHash(Vec<u8>);
 
+impl SessionHash {
+    fn to_hex(&self) -> String {
+        static CHARS: &[char; 16] =
+            &['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'];
+        let mut res = String::with_capacity(2 * self.0.len());
+        for &c in &self.0 {
+            res.push(CHARS[(c & 0b1111) as usize]);
+            res.push(CHARS[(c >> 4) as usize]);
+        }
```

Comment:
I find this a bit arcane 😅 
Maybe add a small test?

Reply:
It's a fair request :+1: